test/Validator.test.ts

```diff
@@ -8,7 +9,9 @@ function readFile(filePath: string) {
 }
 
 function readFileAsBuffer(filePath: string) {
-  return readFile(filePath).buffer;
+  const file = readFile(filePath);
+  const fileBuffer = file.buffer.slice(file.byteOffset, file.byteOffset + file.byteLength);
```

Comment:
I'm not clear on why this was necessary - can you provide an example?

Reply:
See this SO answer from ZachB. The issue with using .buffer directly is that the returned ArrayBuffer from .buffer for files <4kB may contain some 'random' additional data, thus reading data from it could result in getting random bytes which messes with the entire validator.

If you select the FSEQ version V2 Uncompressed correctly in the xLights settings the files won't really get as small as 4kB. But if, for example, you select the FSEQ version V2 ZSTD normal fseq files are mostly below or just slightly bigger than 4kB. As the validator should check that the FSEQ version was set to V2 Uncompressed correctly it needs to be able to read V2 ZSTD files.

The approach with slicing the buffer fixes those problems (this solution is from the SO answer).

Reply:
If you want I can still add an example - but that would involve multiple steps, as it isn't that easy to reproduce.